src/BloomExe/CollectionTab/LibraryListView.cs

```diff
@@ -702,9 +730,11 @@ private void HighlightBookButtonAndShowContextMenuButton(BookInfo bookInfo)
 		{
 			foreach (var btn in AllBookButtons())
 			{
-				if (btn.Tag == bookInfo)
+				var bookButtonInfo = btn.Tag as BookButtonInfo;
+				if (bookButtonInfo.BookInfo == bookInfo)
 				{
-					btn.Paint += btn_Paint;
+					// BL-2678 don't display menu triangle if there's no menu to display
+					if(!bookButtonInfo.IsFactoryTemplate) btn.Paint += btn_Paint;
```

Comment:
Hmmm...now we have yet another place that 'knows' that factory templates have no menu items. Would it be worth adding a method to BookButtonInfo, something like HasNoTriangleMenuItems, so the knowledge that factory templates have this property is in just one place?